ci/fpm-deployment.sh

```diff
@@ -9,7 +9,7 @@ destdir="${DESTDIR:-stdlib-fpm}"
 fypp="${FYPP:-$(which fypp)}"
 
 # Arguments for the fypp preprocessor
-fyflags="${FYFLAGS:--DMAXRANK=4}"
+fyflags="${FYFLAGS:--DMAXRANK=8}"
```

Comment:
I had an issue with one of the tests which has up to 8 dimensions when testing. If I remember correctly is the file test_mean_f03.fypp that when processed creates tests up to dimension 8 for the mean function. Maybe is this file that needs fixing.

Reply:
right, this file is only compiled if `CMAKE_MAXIMUM_RANK` > 7. 
A similar condition should be added in the `fpm-deployment.sh`, e.g. (first and last lines are additions):
```bash
# Arguments for the fypp preprocessor
maxrank=4
fyflags="${FYFLAGS:--DMAXRANK=$maxrank}"

# Number of parallel jobs for preprocessing
if [ $(uname) = "Darwin" ]; then
  njob="$(sysctl -n hw.ncpu)"
else
  njob="$(nproc)"
fi

# Additional files to include
include=(
  "ci/fpm.toml"
  "LICENSE"
  "VERSION"
)

# Files to remove from collection
prune=(
  "$destdir/test/test_always_fail.f90"
  "$destdir/test/test_always_skip.f90"
  "$destdir/test/test_hash_functions.f90"
  "$destdir/src/common.f90"
  "$destdir/src/f18estop.f90"
)

if [[ $maxrank < 8 ]]; then
  prune+=("$destdir/test/stats/test_mean_f03.fypp")
fi
```

Note: I did not test this suggestion.

Reply:
I'll commit your suggestion as this is unrelated to the current PR. But I think it should be looked at as indeed in that file there is a global variable `${t1}$ :: d8_${k1}$(2, 3, 4, 2, 3, 4, 2, 3) = reshape(d1_${k1}$, [2, 3, 4, 2, 3, 4, 2, 3], [${t1}$:: 3])` that causes problems if the MAXRANK=4

Reply:
I'll test and come back

Reply:
That is on purpose. The maximum rank before F03 was 7 and since F03 is 15. Some compilers do (did?) not support up to 15 ranks. Therefore, the compilation of this file depends on maximum rank requested by the user, and supported by the compiler.

Reply:
Done: just had to change [[ $maxrank -lt 8 ]] for [ ${maxrank} -lt 8 ]